From a8dcbd68ea690e85c1b6a924a5d93332cb6eef3b Mon Sep 17 00:00:00 2001
From: hmtheboy154 <buingoc67@gmail.com>
Date: Tue, 23 Jan 2024 14:15:35 +0000
Subject: [PATCH 18/19] bass: remove ntfs install options

We want to only use the Windows installer for those install types
---
 install/scripts/1-install | 38 +++++++++++++++++++++++++++++++++++++-
 1 file changed, 37 insertions(+), 1 deletion(-)

diff --git a/install/scripts/1-install b/install/scripts/1-install
index c7653143..4f636797 100644
--- a/install/scripts/1-install
+++ b/install/scripts/1-install
@@ -227,7 +227,7 @@ convert_fs()
 	fi
 }
 
-format_fs()
+format_fs_old()
 {
 	local cmd
 	echo -e '"Do not re-format" ""\next4 ""\nntfs ""\nfat32 ""\nexfat ""\nf2fs ""' > $menufile
@@ -269,6 +269,42 @@ format_fs()
 	fi
 }
 
+format_fs()
+{
+	local cmd
+	echo -e '"Do not re-format" ""\next4 ""\nexfat ""\nf2fs ""' > $menufile
+	set_answer_if_auto $FORCE_FORMAT
+	choose "Choose filesystem" "Please select a filesystem to format $1:"
+	case "$choice" in
+		ext4)
+			cmd="mkfs.ext3 -L"
+			;;
+		exfat)
+			cmd="mkfs.exfat -L"
+			;;
+		f2fs)
+			cmd="make_f2fs -l"
+			;;
+		*)
+			;;
+	esac
+	if [ -n "$cmd" ]; then
+		[ -n "$OEM_INSTALL" ] || dialog --title " Confirm " --defaultno --yesno \
+			"\n You chose to format $1 to $choice.\n All data in that partition will be LOST.\n\n Are you sure to format the partition $1?" 10 59
+		[ $? -ne 0 ] && return 1
+		$cmd Android-x86 /dev/$1 | awk '{
+			# FIXME: very imprecise progress
+			if (match($0, "done"))
+				printf("%d\n", i+=33)
+		}' | progress_bar "Formatting" "Formatting partition $1..."
+		convert_fs $1
+	elif blkid /dev/$1 | grep -q ext[23]; then
+		dialog --clear --title " Warning " --yesno \
+			"\nYou chose to install android-x86 to an ext2/3 filesystem. We suggest you convert it to ext4 for better reliability and performance." 9 62
+		[ $? -eq 0 ] && convert_fs $1
+	fi
+}
+
 create_winitem()
 {
 	win=`fdisk -l /dev/$(echo $1 | cut -b-3) | grep ^/dev | cut -b6-12,55- | awk '{
-- 
2.34.1

